ts/features/idpay/barcode/components/BarcodeExpirationSlider.tsx

```diff
@@ -0,0 +1,58 @@
+import { LabelSmall, VSpacer } from "@pagopa/io-app-design-system";
+import * as React from "react";
+import { StyleSheet, View } from "react-native";
+import I18n from "../../../../i18n";
+import { ProgressBar } from "../../../bonus/bpd/screens/details/components/summary/base/ProgressBar";
+
+export const IdPayBarcodeExpireSlider = ({
+  secondsToExpiration,
+  secondsExpirationTotal
+}: {
+  secondsToExpiration: number;
+  secondsExpirationTotal: number;
+}) => {
+  const [seconds, setSeconds] = React.useState(secondsToExpiration);
+  const isCodeExpired = seconds === 0;
+  React.useEffect(() => {
+    const timer = setInterval(() => {
+      setSeconds(currentSecs => currentSecs - 1);
+    }, 1000);
+    if (seconds <= 0) {
+      setSeconds(0);
+      clearInterval(timer);
+    }
+    return () => clearInterval(timer);
+    // possible over-engineering, but we actually
+    // only need to run zero checks once the code is expired,
+    // there is no reason to rerun this hook every second
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isCodeExpired]);
+
+  const formattedExpireMinutesString = isCodeExpired
+    ? ""
+    : new Date(seconds * 1000).toISOString().slice(14, 19);
```

Comment:
I think this would be nicer (and safer). What do you think?

```suggestion
const formattedExpireMinutesString = !isCodeExpired && format(new Date(seconds * 1000), "mm:ss");
```

Reply:
approved, even though I don't really like passing a possible boolean to render a string, so I suggest using your format function alongside a ternary operator; what do you think? 
```ts
  const formattedExpireMinutesString = isCodeExpired
    ? ""
    : format(new Date(seconds * 1000), "mm:ss");
```